### PR TITLE
Add everything so far

### PR DESCRIPTION
**Pull request by A248**
The kitpvp plugin codebase is mostly done.

What is not yet complete:
* Stats GUI or command
* /kitpvp-admin createkit and deletekit, pending https://github.com/SolarMC-Dev/DataLoader/issues/32
* Placeholders; I will be depending on PlaceholderAPI just like the codebase I inherited did
* Run existing tests and add more tests

### CODE REVIEW

File: src/main/java/gg/solarmc/kitpvp/handler/KillHandler.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+                }
+            }
+
+            BigDecimal killReward = violence().killRewards().killReward().findValue(killerNewKillstreak);
+            DepositResult depositResult = killer.getSolarPlayer().getData(CreditsKey.INSTANCE)
+                    .depositBalance(tx, killReward);
+            postTransactCallbacks.add(mainKillReward(killer, victimName, killReward, depositResult));
+
+        }).thenRun(() -> {
+            for (Runnable callback : postTransactCallbacks) {
```
Comment (auriium) [inline src/main/java/gg/solarmc/kitpvp/handler/KillHandler.java]: Didnt think of this

Comment (auriium) [reply src/main/java/gg/solarmc/kitpvp/handler/KillHandler.java]: Preserves code readability reasonably though - why is this in thenRun though

Comment (auriium) [reply src/main/java/gg/solarmc/kitpvp/handler/KillHandler.java]: shouldn't be thenRunSync?

File: src/main/java/gg/solarmc/kitpvp/listeners/KillListener.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    public void reload() {
+        damageTrackers = Caffeine.newBuilder().weakKeys()
+                .expireAfterAccess(Duration.ofMinutes(configCenter.config().violence().damageExpireMinutes()))
+                .build();
+    }
+
+    @Override
+    public void stop() { }
+
+    private class DamageTracker {
```
Comment (auriium) [inline src/main/java/gg/solarmc/kitpvp/listeners/KillListener.java:87]: record?

Comment (auriium) [reply src/main/java/gg/solarmc/kitpvp/listeners/KillListener.java:87]: thank you for using weakKeys

File: src/main/java/gg/solarmc/kitpvp/misc/FuturePoster.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * You should have received a copy of the GNU Affero General Public License
+ * along with kitpvp. If not, see <https://www.gnu.org/licenses/>
+ * and navigate to version 3 of the GNU Affero General Public License.
+ */
+
+package gg.solarmc.kitpvp.misc;
+
+import space.arim.omnibus.util.concurrent.CentralisedFuture;
+
+public interface FuturePoster {
```
Comment (auriium) [inline src/main/java/gg/solarmc/kitpvp/misc/FuturePoster.java:24]: Is this a class for explicitly handling the tail ends of futures?

**Review by auriium — COMMENTED** [review]
great work.

File: src/main/java/gg/solarmc/kitpvp/listeners/KillListener.java (hunk 2 above)
Comment (A248) [reply src/main/java/gg/solarmc/kitpvp/listeners/KillListener.java:87]: The class has mutable fields and is therefore not a record

File: src/main/java/gg/solarmc/kitpvp/misc/FuturePoster.java (hunk 3 above)
Comment (A248) [reply src/main/java/gg/solarmc/kitpvp/misc/FuturePoster.java:24]: Yes

File: src/main/java/gg/solarmc/kitpvp/handler/KillHandler.java (hunk 1 above)
Comment (A248) [reply src/main/java/gg/solarmc/kitpvp/handler/KillHandler.java]: Thanks to https://github.com/SolarMC-Dev/SolarPaper-API/commit/fb2d80601e5b164f97f79b6924c4f777c447a849 sendMessage with Components is guaranteed to be thread-safe (and indeed the implementation is)

File: src/main/java/gg/solarmc/kitpvp/listeners/KillListener.java (hunk 2 above)
Comment (auriium) [reply src/main/java/gg/solarmc/kitpvp/listeners/KillListener.java:87]: Was asking if you wanted to make it one

**Pull request merged**